src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -19,35 +21,68 @@ public class DeleteCommand extends Command {
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the client identified by the index number used in the displayed client list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the client identified by the index number or name used in the displayed client list.\n"
+            + "Parameters: INDEX (must be a positive integer) or " + PREFIX_NAME + "NAME \n"
+            + "Example: " + COMMAND_WORD + " 1 or " + COMMAND_WORD + " " + PREFIX_NAME + "John Doe";
+
 
     public static final String MESSAGE_DELETE_HELP = "Delete Command: (Coming soon)\n"
             + "Format: (Coming soon)\n";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Client: %1$s";
 
     private final Index targetIndex;
+    private final Name targetName;
 
+    /**
+     * Creates a DeleteCommand to delete the person at the specified {@code Index}
+     */
     public DeleteCommand(Index targetIndex) {
         this.targetIndex = targetIndex;
+        this.targetName = null;
+    }
+
+    /**
+     * Creates a DeleteCommand to delete the person with the specified {@code Name}
+     */
+    public DeleteCommand(Name targetName) {
+        this.targetName = targetName;
+        this.targetIndex = null;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
+        Person personToDelete;
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        if (targetIndex != null) {
+            personToDelete = deleteByIndex(lastShownList);
+        } else {
+            personToDelete = deleteByName(lastShownList);
         }
 
-        Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
         model.deletePerson(personToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
     }
 
+    private Person deleteByIndex(List<Person> lastShownList) throws CommandException {
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        return lastShownList.get(targetIndex.getZeroBased());
+    }
+
+    private Person deleteByName(List<Person> lastShownList) throws CommandException {
+        for (Person person : lastShownList) {
+            if (person.getName().equals(targetName)) {
+                return person;
+            }
+        }
+        throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_NAME);
+    }
+
```

Comment:
Can consider doing this in another class like `DeleteCommandExecutor` meant just for deletion logic

Reply:
agree, SLAP

Reply:
Done!